src/Extensions/Compatability/Validation/Code.cs

```diff
@@ -0,0 +1,489 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.System.Validation;
+
+namespace Microsoft.Omex.Extensions.Compatability.Validation
+{
+	/// <summary>
+	/// Code validation
+	/// </summary>
+	public static class Code
+	{
+		/// <summary>
+		/// Check object's state, throws specified Exception type if state evaluates to false.
+		/// </summary>
+		/// <typeparam name="TException">Exception to throw</typeparam>
+		/// <param name="state">Object's state</param>
+		/// <param name="errorMessage">Error message</param>
+		/// <param name="tagId">Tag Id to log, leave null if no logging is needed</param>
+		public static void Expects<TException>(bool state, string errorMessage, EventId? tagId)
+			where TException : Exception
+		{
+			if (!Validate(state, errorMessage, tagId))
+			{
+				ReportError<TException>(errorMessage);
+			}
+		}
+
+
+		/// <summary>
+		/// Checks that the enumerable argument is not null and doesn't contain any nulls
+		/// </summary>
+		/// <remarks>Be careful to not pass enumerables that can be enumerated only once</remarks>
+		/// <typeparam name="T">Type of the enumerable</typeparam>
```

Comment:
>T [](start = 23, length = 1)

Don't use `T` as per C# best practices. It's hard to work out what this means exactly, so something like `TEnumerable` would be best. Applies throughout this class.

Reply:
Fixed, thanks, just for a context it is a copy of the code that I'll really like to deprecate